lib/timescale/acts_as_hypertable.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Timescale
+  # If you want your model to hook into its underlying hypertable
+  # as well as have access to TimescaleDB specific data, methods, and more,
+  # specify this macro in your model.
+  #
+  # @note Your model's table needs to have already been converted to a hypertable
+  # via the TimescaleDB `create_hypertable` function for this to work.
+  #
+  # @see https://docs.timescale.com/api/latest/hypertable/create_hypertable/ for
+  #   how to use the SQL `create_hypertable` function.
+  # @see Timescale::MigrationHelpers#create_table for how to create a new hypertable
+  # via a Rails migration utilizing the standard `create_table` method.
+  #
+  # @example Enabling the macro on your model
+  #   class Event < ActiveRecord::Base
+  #     acts_as_hypertable
+  #   end
+  #
+  # @see Timescale::ActsAsHypertable::ClassMethods#acts_as_hypertable
+  #   for configuration options
+  module ActsAsHypertable
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      # == Configuration options
+      #
+      # @param [Hash] options The options to initialize your macro with.
+      # @option options [Symbol] :time_column The name of the column in your
+      #   model's table containing time values. The name provided should be
+      #   the same name as the `time_column_name` you passed to the
+      #   TimescaleDB `create_hypertable` function.
+      #
+      # @example Enabling the macro on your model with options
+      #     class Event < ActiveRecord::Base
+      #       acts_as_hypertable time_column: :timestamp
+      #     end
+      #
+      def acts_as_hypertable(options = {})
+        return if included_modules.include?(Timescale::ActsAsHypertable::HyperableInstanceMethods)
+
+        extend Timescale::ActsAsHypertable::HyperableClassMethods
+        include Timescale::ActsAsHypertable::HyperableInstanceMethods
+
+        class_attribute :hypertable_options, instance_writer: false
+
+        self.hypertable_options = { time_column: :created_at }
```

Comment:
❤️

Reply:
I think we could also introduce like some defaults for the entire rails app. WDYT?
```suggestion
        self.hypertable_options = Timescale.default_hypertable_options
```

Reply:
I like this idea! Will add it as I implement your other suggestions 😄